core/charm/repository/charmstore.go

```diff
@@ -135,6 +139,47 @@ func (c *CharmStoreRepository) ListResources(charmURL *charm.URL, _ corecharm.Or
 	return nil, nil
 }
 
+// GetEssentialMetadata resolves each provided MetadataRequest and returns back
+// a slice with the results. The results include the minimum set of metadata
+// that is required for deploying each charm.
+func (c *CharmStoreRepository) GetEssentialMetadata(reqs ...corecharm.MetadataRequest) ([]corecharm.EssentialMetadata, error) {
+	var res = make([]corecharm.EssentialMetadata, len(reqs))
+
+	for reqIdx, req := range reqs {
+		// NOTE(achilleas): due to the way that the charmstore client
+		// was originally implemented we unfortunately need to create a
+		// new client per request.
+		channel := csparams.Channel(req.Origin.Channel.Risk)
+		client, err := c.clientFactory(c.charmstoreURL, channel, req.Macaroons)
+
+		if _, err = client.Meta(req.CharmURL, &res[reqIdx]); err != nil {
+			return nil, errors.Annotatef(err, "retrieving metadata for %q", req.CharmURL)
+		}
+		res[reqIdx].ResolvedOrigin = req.Origin
+
+		// The metadata call does not return back LXD profile
+		// information.  We need to make a separate call to grab it
+		// from within the archive.
+		profileReader, fetchErr := client.GetFileFromArchive(req.CharmURL, "lxd-profile.yaml")
+		if fetchErr != nil {
+			// It's fine if the charm does not provide an lxd profile
+			if errors.Cause(fetchErr) == params.ErrNotFound {
+				continue
+			}
+
+			return nil, errors.Annotatef(err, "retrieving metadata for %q", req.CharmURL)
+		}
+
+		res[reqIdx].LXDProfile, err = charm.ReadLXDProfile(profileReader)
+		_ = profileReader.Close()
```

Comment:
The return from the method should be used at least to inform the user that something went wrong there.

Reply:
`charm.ReadLXDProfile` is expected to consume the entire profile payload and either return back an error or return a profile instance. In this scenario, failing to close the reader (which in reality is an http.Response.Body) shouldn't really matter as the actual goal of the method (grab the metadata) has already been met at this point. 

We could however catch the error and log out a warning if you think it's important. Thoughts?